### DISCUSSION TITLE
Loading custom models

### DISCUSSION THREAD

**Discussion by NoDowt** [5axes/Calibration-Shapes#98] [Ideas]
I know we can change the STL files to use different models for the predefined items in the plugin; however, would it be possible to have your plugin dynamically load STL? e.g. at cura load, the plugin checks a folder (e.g. models/custom) and then adds a menu item for each STL file in that folder?

Could be nice to have more temp tower options, without you needing to maintain them or us manually changing the predefined files around (or loading STLs manually of course). And other might have different calibration models they use frequently they could add also.

**Comment by 5axes** [comment]
This is an idea I had at the beginning of the creation of this plugin.
But I abandoned it, firstly just having a menu to list user stl files, didn't seem to me to bring much added value to a plugin.
Secondly when updating a plugin the installation procedure first deletes the existing folder completely and then recreates it. So there is a strong risk for the user to lose all his files.
This is not a plugin I would use but maybe a tool to load the current files of a user with a storage in a folder independent of the plugin could be a good idea.  Something to think about for the future ...
